test/conformance/kuttl/mutate/refactor/add-volume-deployment/policy.yaml

```diff
@@ -0,0 +1,46 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: add-volume
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: add-volume
+  annotations:
+    policies.kyverno.io/title: Add Volume to Deployment
+    policies.kyverno.io/category: Sample
+    policies.kyverno.io/subject: Deployment, Volume
+    policies.kyverno.io/minversion: 1.6.0
+    policies.kyverno.io/description: >-
+      Some Kubernetes applications like HashiCorp Vault must perform some modifications
+      to resources in order to invoke their specific functionality. Often times, that functionality
+      is controlled by the presence of a label or specific annotation. This policy, based on HashiCorp
+      Vault, adds a volume and volumeMount to a Deployment if there is an annotation called
+      "vault.k8s.corp.net/inject=enabled" present.
+spec:
+  rules:
+  - name: add-volume
+    match:
+      any:
+      - resources:
+          kinds:
+          - Deployment
+    preconditions:
+      any:
+      - key: "{{request.object.spec.template.metadata.annotations.\"vault.k8s.corp.net/inject\"}}"
+        operator: Equals
+        value: enabled
+    mutate:
+      patchesJson6902: |-
+        - op: add
+          path: /spec/template/spec/volumes/-
+          value:
+            name: vault-secret
+            emptyDir:
+              medium: Memory
+        - op: add
+          path: /spec/template/spec/containers/0/volumeMounts/-
```

Comment:
@chipzoller you're only mutating the first container here

Reply:
hmm, looks correct, my bad.

Reply:
Yeah, this is just what the current policy has defined.

Reply:
The problem is with text tear down, takes too long to delete the pods...

Reply:
Looks like a bug in k8s, i added a small sleep and it seems to work fine now.